storage/innobase/handler/ha_innodb.cc

```diff
@@ -8002,7 +8002,7 @@ ha_innobase::write_row(
 		error, m_prebuilt->table->flags, m_user_thd);
 
 #ifdef WITH_WSREP
-	if (!error_result && trx->is_wsrep()
+	if (!error_result && wsrep_on(m_user_thd)
 	    && !trx->is_bulk_insert()
 	    && wsrep_thd_is_local(m_user_thd)
 	    && !wsrep_thd_ignore_table(m_user_thd)
```

Comment:
Can we consolidate more checks into one function call, and can we avoid introducing a mandatory function call for the case where Galera replication is not being used? The `trx->is_wsrep()` was an inline check for `trx->wsrep`.

This `if` expression is also invoking `thd_sql_command()` twice.